src/libs/actions/NetworkRequestQueue.js

```diff
@@ -1,15 +1,48 @@
 import Onyx from 'react-native-onyx';
+import _ from 'underscore';
+import lodashUnionWith from 'lodash/unionWith';
 import ONYXKEYS from '../../ONYXKEYS';
 
+const retryMap = new Map();
+let persistedRequests = [];
+
+Onyx.connect({
+    key: ONYXKEYS.NETWORK_REQUEST_QUEUE,
+    callback: val => persistedRequests = val || [],
+});
+
 function clearPersistedRequests() {
     Onyx.set(ONYXKEYS.NETWORK_REQUEST_QUEUE, []);
+    retryMap.clear();
 }
 
 function saveRetryableRequests(retryableRequests) {
-    Onyx.merge(ONYXKEYS.NETWORK_REQUEST_QUEUE, retryableRequests);
+    persistedRequests = lodashUnionWith(persistedRequests, retryableRequests, _.isEqual);
```

Comment:
This change has introduced this [bug](https://github.com/Expensify/App/issues/14584). (Title: pinned chat become unpinned when user pinned chat in offline mode)

If a user takes an action that is identical(_.isEqual) to older action then it will be removed.

To fix the issue we have replace above code by:
`
persistedRequests = persistedRequests.concat(requestsToPersist);
`
The PR that fixes the issue: https://github.com/Expensify/App/pull/14608

Reply:
The idea is to prevent the user from accumulate duplicate requests while offline 
If you perform the same action 10x while offline, we skip posting 10 Network request but just one 
This is an optimization that helps users in low network conditions to not waste bandwidth

From the information shared here it seems that pinning and unpinning the chat uses the same command / request, I would suggest to make them distinct actions so that `_.isEqual` can differentiate them correctly

Reply:
We shouldn't have done this. Context https://expensify.slack.com/archives/C01GTK53T8Q/p1685822789307339